tests/Unit/PrestaShopBundle/Form/Admin/Sell/Order/Invoices/InvoiceOptionsDataProviderTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\PrestaShopBundle\Form\Admin\Sell\Order\Invoices;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShopBundle\Form\Admin\Sell\Order\Invoices\InvoiceOptionsDataProvider;
+use PrestaShopBundle\Form\Admin\Sell\Order\Invoices\InvoiceOptionsType;
+use PrestaShopBundle\Form\Exception\DataProviderException;
+
+class InvoiceOptionsDataProviderTest extends TestCase
+{
+    /**
+     * Tests that exception is thrown if invoice number is incorrect
+     */
+    public function testValidateFailsOnIncorrectInvoiceNumber(): void
+    {
+        $dataConfiguration = $this->getMockBuilder(DataConfigurationInterface::class)->getMock();
+        $invoiceOptionsDataProvider = new InvoiceOptionsDataProvider($dataConfiguration, 5);
+        $data = [
+            InvoiceOptionsType::FIELD_INVOICE_NUMBER => 4,
+        ];
+        $this->expectException(DataProviderException::class);
+        $invoiceOptionsDataProvider->setData($data);
+    }
+
+    /**
+     * Tests that all 3 text fields throw an exception if you pass HTML
+     *
+     * @dataProvider getInvoiceNoHtmlFields
+     *
+     * @param string $field
+     */
+    public function testValidateFailsOnHtmlTags(string $field): void
+    {
+        $dataConfiguration = $this->getMockBuilder(DataConfigurationInterface::class)->getMock();
+        $invoiceOptionsDataProvider = new InvoiceOptionsDataProvider($dataConfiguration, 5);
+        $data = [
+            $field => [
+                1 => 'text',
+                2 => '<html>',
+            ],
+        ];
+        $this->expectException(DataProviderException::class);
```

Comment:
I think you should not only expect the exception but also check the content to be sure the error collection is correctly filled You can catch the error and assert its type and content to do that